File: 0775-n-ary-tree-preorder-traversal/0775-n-ary-tree-preorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    List<Integer> res = new ArrayList<>();
    private void traverse(Node root) {
        if (root == null) {
            return;
        }
        res.add(root.val);
        for (Node child: root.children) {
            traverse(child);
        }
    }
    public List<Integer> preorder(Node root) {
        traverse(root);
        return res;       
    }
}

File: 0775-n-ary-tree-preorder-traversal/0775-n-ary-tree-preorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def traverse(self, root: 'Node', res: List[int]) -> None:
        if not root:
            return
        res.append(root.val)
        for child in root.children:
            self.traverse(child, res)

    def preorder(self, root: 'Node') -> List[int]:
        res = []
        self.traverse(root, res)
        return res